testclient/mainwindow.cpp

```diff
@@ -243,15 +251,10 @@ void MainWindow::printPBClicked()
 
     case DISCONNECTED:
         QMessageBox::information(this, tr("Error"), tr("Not Connected To a Printer"));
-        if(!core->pluginLoaded()){
-            addLog(tr("No plugin loaded !"));
-            addLog(tr("Requesting Firmware..."));
-            core->requestFirmware();
-        }
         break;
```

Comment:
The problem is: 
If we have a serial connection done, the state is DISCONNECTED and not CONNECTING.
In some place we need to check the serial is connected, and set CONNECTING.

Other thing..
`void MainWindow::connectPBClicked()
{
    if (core->state() == DISCONNECTED) {
        core->initFirmware(ui->serialPortCB->currentText(), ui->baudRateLE->text().toInt());
        connect(core->serial(), &SerialLayer::receivedCommand, this, &MainWindow::checkReceivedCommand);
        connect(core->serial(), &SerialLayer::pushedCommand, this, &MainWindow::checkPushedCommands);
        addLog(tr("Serial connected"));
        ui->connectPB->setText(tr("Disconnect"));
        if (!core->pluginLoaded()) {
            addLog(tr("No plugin loaded !"));
            addLog(tr("Requesting Firmware..."));
            core->requestFirmware();
        }
    } else {
        core->serial()->closeConnection();
        core->setState(DISCONNECTED);
        addLog(tr("Disconnected"));
        ui->connectPB->setText(tr("Connect"));
    }
}`

If we are in CONNECTING and we click in printPB it will change to DISCONNECTED ? Do not make sense in this case.